### PR TITLE
CORCI-960 build: Update MACSio to build an openmpi3 package

### PR DESCRIPTION
**Pull request by phender**
Update the MACSio spec file to build the current package as a
macsio-mpich and add building a macsio-openmpi3 package

Signed-off-by: Phillip Henderson <phillip.henderson@intel.com>

### CODE REVIEW

File: MACSio.spec [hunk 1]
```diff
@@ -45,32 +58,74 @@ We hope MACSio helps to put the MAX in scalable I/O performance ;)
...
 %endif
-cmake -DCMAKE_INSTALL_PREFIX=%{_bindir} \
-    -DWITH_JSON-CWX_PREFIX=/usr \
+for mpi in %{?mpi_list}
+do
+  mkdir $mpi
+  pushd $mpi
+  %module_load $mpi
+  cmake -DCMAKE_INSTALL_PREFIX=%{_bindir}/$mpi \
+    -DWITH_JSON-CWX_PREFIX=%{_usr} \
```
Comment (brianjmurrell) [inline MACSio.spec]: This should probably be:
```suggestion
    -DWITH_JSON-CWX_PREFIX=%{prefix} \
```

File: MACSio.spec [hunk 2]
```diff
@@ -45,32 +58,74 @@ We hope MACSio helps to put the MAX in scalable I/O performance ;)
...
     -DENABLE_SILO_PLUGIN=OFF \
     -DENABLE_HDF5_PLUGIN=ON \
-    -DWITH_HDF5_PREFIX=/usr/lib64/mpich
-%if (0%{?suse_version} >= 1500)
-sed -i -e s/H5pubconf.h/H5pubconf-64.h/ plugins/macsio_hdf5.c
-%endif
-make
+    -DWITH_HDF5_PREFIX=%{_libdir}/$mpi \
+    ..
+  make
```
Comment (brianjmurrell) [inline MACSio.spec]:
```suggestion
%{make_build}
```

File: MACSio.spec [hunk 3]
```diff
@@ -45,32 +58,74 @@ We hope MACSio helps to put the MAX in scalable I/O performance ;)
...
+  sed -i -e s/H5pubconf.h/H5pubconf-64.h/ plugins/macsio_hdf5.c
 %endif
-cmake -DCMAKE_INSTALL_PREFIX=%{_bindir} \
-    -DWITH_JSON-CWX_PREFIX=/usr \
+for mpi in %{?mpi_list}
+do
+  mkdir $mpi
+  pushd $mpi
+  %module_load $mpi
+  cmake -DCMAKE_INSTALL_PREFIX=%{_bindir}/$mpi \
```
Comment (brianjmurrell) [inline MACSio.spec]: Isn't the typical mulit-mpi-variant model to install things into `%{_libdir}/$mpi/{bin,lib,etc.}`?  So:
```suggestion
  cmake -DCMAKE_INSTALL_PREFIX=%{_libdir}/$mpi \
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: MACSio.spec (hunk 1 above)
Comment (phender) [reply MACSio.spec]: Done

File: MACSio.spec (hunk 2 above)
Comment (phender) [reply MACSio.spec]: Done

File: MACSio.spec (hunk 3 above)
Comment (phender) [reply MACSio.spec]: Done

File: MACSio.spec [hunk 4]
```diff
@@ -93,13 +93,13 @@ do
   mkdir $mpi
   pushd $mpi
   %module_load $mpi
-  cmake -DCMAKE_INSTALL_PREFIX=%{_bindir}/$mpi \
-    -DWITH_JSON-CWX_PREFIX=%{_usr} \
+  cmake -DCMAKE_INSTALL_PREFIX=%{_libdir}/$mpi \
```
Comment (brianjmurrell) [inline MACSio.spec]: Hrm.  Interesting.  This is not actually an install _prefix_ but rather is the actual directory to install the single produced binary into.  So in that case, I think you actually want:

```suggestion
  cmake -DCMAKE_INSTALL_PREFIX=%{_libdir}/$mpi/bin \
```

Sorry.  I didn't realise how basic this package was and just assumed that was a _prefix_.

File: MACSio.spec [hunk 5]
```diff
@@ -116,11 +116,11 @@ done
 %license LICENSE
 %if %{with_mpich}
 %files mpich
-%{_bindir}/mpich/*
+%{_libdir}/mpich/*
```
Comment (brianjmurrell) [inline MACSio.spec]:
```suggestion
%{_libdir}/mpich/bin/*
```

File: MACSio.spec [hunk 6]
```diff
@@ -116,11 +116,11 @@ done
...
 %endif
 %if %{with_openmpi3}
 %files openmpi3
-%{_bindir}/openmpi3/*
+%{_libdir}/openmpi3/*
```
Comment (brianjmurrell) [inline MACSio.spec]:
```suggestion
%{_libdir}/openmpi3/bin/*
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: MACSio.spec (hunk 4 above)
Comment (phender) [reply MACSio.spec]: Done

File: MACSio.spec (hunk 5 above)
Comment (phender) [reply MACSio.spec]: Done

File: MACSio.spec (hunk 6 above)
Comment (phender) [reply MACSio.spec]: Done

**Comment by phender** [comment]
Started https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2FMACSio/detail/PR-2/12 to verify compatibility with hdf5 1.12 now that https://github.com/daos-stack/hdf5/pull/9 is merged.

**Comment by phender** [comment]
> Started https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2FMACSio/detail/PR-2/12 to verify compatibility with hdf5 1.12 now that [daos-stack/hdf5#9](https://github.com/daos-stack/hdf5/pull/9) is merged.

I need to wait for https://build.hpdd.intel.com/job/daos-stack/job/hdf5/job/master/10/ to complete

**Comment by phender** [comment]
> > Started https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2FMACSio/detail/PR-2/12 to verify compatibility with hdf5 1.12 now that [daos-stack/hdf5#9](https://github.com/daos-stack/hdf5/pull/9) is merged.
> 
> I need to wait for https://build.hpdd.intel.com/job/daos-stack/job/hdf5/job/master/10/ to complete

https://build.hpdd.intel.com/job/daos-stack/job/MACSio/job/PR-2/13/ started now that https://build.hpdd.intel.com/job/daos-stack/job/hdf5/job/master/10/ is complete.

**Comment by phender** [comment]
> > > Started https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2FMACSio/detail/PR-2/12 to verify compatibility with hdf5 1.12 now that [daos-stack/hdf5#9](https://github.com/daos-stack/hdf5/pull/9) is merged.
> > 
> > 
> > I need to wait for https://build.hpdd.intel.com/job/daos-stack/job/hdf5/job/master/10/ to complete
> 
> https://build.hpdd.intel.com/job/daos-stack/job/MACSio/job/PR-2/13/ started now that https://build.hpdd.intel.com/job/daos-stack/job/hdf5/job/master/10/ is complete.

Verified build is successful with hdf5 1.12 packages: https://build.hpdd.intel.com/job/daos-stack/job/MACSio/job/PR-2/13/artifact/artifacts/centos7/installed_pkgs.log

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**